src/scipp/signal/__init__.py

```diff
@@ -0,0 +1,107 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+"""Sub-package for signal processing such as smoothing.
+
+This subpackage provides wrappers for a subset of functions from
+:py:mod:`scipy.signal`.
+"""
+from dataclasses import dataclass
+from numpy import ndarray
+
+from ..core import array, identical, islinspace, to_unit, DataArray, Variable
+from ..core import UnitError, CoordError
+from ..units import one
+from ..compat.wrapping import wrap1d
+
+
+@dataclass
+class SOS:
+    """
+    Second-order sections representation returned by :py:func:`scipp.signal.butter`.
+
+    This is used as input to :py:func:`scipp.signal.sosfiltfilt`.
+    """
+    coord: Variable
+    sos: ndarray
+
+
+def _frequency(coord: Variable) -> Variable:
+    if not islinspace(coord).value:
+        raise CoordError("Data is not regularly sampled, cannot compute frequency.")
+    return (len(coord) - 1) / (coord[-1] - coord[0])
+
+
+def butter(da: DataArray, dim: str, *, N, Wn, **kwargs):
+    """
+    Butterworth digital and analog filter design.
+
+    Design an Nth-order digital or analog Butterworth filter and return the filter
+    coefficients.
+
+    This is intended for use with :py:func:`scipp.signal.sosfiltfilt`. See there for
+    an example.
+
+    This is a wrapper around :py:func:`scipy.signal.butter`. See there for a
+    complete description of parameters. The differences are:
+
+    - Instead of a sampling frequency fs, this wrapper takes a data array ``da`` and a
+      dimension label ``dim`` as input. The sampling frequency is then computed from
+      ``da.coords[dim]``. Only inputs with regular sampling are supported.
```

Comment:
```suggestion
      ``da.coords[dim]``. Only inputs with fixed sampling are supported.
```
?
Logarithmically spaced coords are also 'regular'.

Reply:
Changed to "Only data sampled at regular intervals are supported"